fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/pdagg/PushDownAggregateCollector.java

```diff
@@ -568,4 +574,104 @@ private int groupByCardinality(ColumnStatistic statistic, double rowCount) {
 
         return 2;
     }
+
+    private boolean isSmallBroadcastJoin(OptExpression optExpression) {
+        if (!sessionVariable.isCboPushDownAggregateOnBroadcastJoin()) {
+            return false;
+        }
+
+        Statistics rightStatistics = optExpression.inputAt(1).getStatistics();
+        if (rightStatistics == null) {
+            return false;
+        }
+        double rightRows = rightStatistics.getOutputRowCount();
+        return rightRows <= sessionVariable.getBroadcastRowCountLimit() &&
+                rightRows <= StatisticsEstimateCoefficient.SMALL_BROADCAST_JOIN_ROW_COUNT_UPPER_BOUND;
```

Comment:
Why we can't use `getBroadcastRowCountLimit` directly?

Reply:
Because `getBroadcastRowCountLimit ` is to large, whose default values is 15000000.
And we only apply this optimization to the very small broadcast join (<= 4096).